src/app/pages/apps/components/available-apps/available-apps-header/available-apps-header.component.ts

```diff
@@ -116,7 +129,31 @@ export class AvailableAppsHeaderComponent implements OnInit {
     dialogRef.componentInstance.submit();
     dialogRef.componentInstance.success.pipe(untilDestroyed(this)).subscribe(() => {
       this.dialogService.closeAllDialogs();
-      this.refreshAvailableApps$.next(true);
+      this.applicationsStore.initialize();
```

Comment:
The `initialize` method is called inside the store in the constructor, this seems redundant.

`refreshAvailableApps$` is used as a trigger to refresh available apps, catalogs, etc.

Reply:
The dialog makes the `catalog.sync_all` call, which might change the catalogs installed and the apps available on those catalogs. So, refreshing the store state from middleware makes sense here.